Setting up using WSL Added (#131)
kroxylicious/kroxylicious: README.md

Kroxylicious is an exploration for building a Kafka protocol proxy,
addressing use cases such as multi-tenancy, schema validation, or encryption.

# Setting Up in Windows Using WSL
### For installing WSL (Skip the First '4' steps if already done)
1. Enable the Windows Subsystem for Linux feature: To enable WSL, you need to enable the Windows Subsystem for Linux feature in the Windows Features dialog.

2. Install a Linux distribution from the Microsoft Store. The rest of these instructions assume a distribution (such as Ubuntu) which uses `apt` for package management, but the process should be similar for distributions using other package managers/ such as Fedora/`dnf`.

3. Launch the Linux distribution and Configure the Linux environment: After launching the Linux distribution, you can configure the environment by creating a user account and setting the password.

With these steps, you should now have WSL installed and configured on your Windows system. 

For further assistance please see the [Microsoft documentation](https://learn.microsoft.com/en-us/windows/wsl/install)

4. Open the WSL window.

> 5.a. Install Latest Java version by typing 

Reviewer: nit: This isn't a sublist technically speaking and probably won't render as part of the list.

Author: okay I have separated those , do let me know if any changes needed plus I have decide to work with this organization after roaming here and there so its still left to involve in this right?

```bash
java --version
```
and Choose the version (preferably v18), it should look like:
```bash

> sudo apt install openjdk-18-jre-headless

Reviewer: I think this is part of 5.b. after the `apt update`

Kroxylicious is an exploration for building a Kafka protocol proxy,
addressing use cases such as multi-tenancy, schema validation, or encryption.

# Setting Up in Windows Using WSL

> ### For installing WSL (Skip the First '3' steps if already done)

Reviewer: ```suggestion
### For installing WSL (Skip the First '4' steps if already done)
```

Or move step `4.` to be part of what is currently `5.b`

1. Enable the Windows Subsystem for Linux feature: To enable WSL, you need to enable the Windows Subsystem for Linux feature in the Windows Features dialog.

> 2. Install a Linux distribution from the Microsoft Store(Ubuntu preferably).

Reviewer 2: ```suggestion
2. Install a Linux distribution from the Microsoft Store. The rest of these instructions assume a distribution (such as Ubuntu) which uses `apt` for package management, but the process should be similar for distributions using other package managers/ such as Fedora/`dnf`.
```

3. Launch the Linux distribution and Configure the Linux environment: After launching the Linux distribution, you can configure the environment by creating a user account and setting the password.

With these steps, you should now have WSL installed and configured on your Windows system. 

> For further assistance:https://learn.microsoft.com/en-us/windows/wsl/install

Reviewer: ```suggestion
For further assistance please see the [Microsoft documentation](https://learn.microsoft.com/en-us/windows/wsl/install)
```

4. Open the WSL window and update the packages using

```bash
  sudo apt update
  sudo apt upgrade
```
5. Install Latest Java version by typing 

> ```bash

Reviewer: This should probably be a [sublist](https://www.markdownguide.org/basic-syntax/#lists-1)

> java

Reviewer: ```suggestion
java --version
```

```
and Choose the version (preferably v18), it should look like:
```bash
sudo apt install openjdk-18-jre-headless
```

> 6. Check whether Java is installed correctly or not using:

Reviewer: This is really covered already in step 5

$ dot -Tpng kroxylicious/target/generated-sources/annotations/deptective.dot > kroxylicious-arch.png
```

# Setting Up in Windows Using WSL
### For installing WSL (Skip this step if already done)
1. Enable the Windows Subsystem for Linux feature: To enable WSL, you need to enable the Windows Subsystem for Linux feature in the Windows Features dialog.

> 2. Install a Linux distribution from the Microsoft Store(Ubuntu preferably)

Reviewer: Is there a particular reason to prefer Ubuntu?

Author: User-friendliness, Frequent Releases: Ubuntu is released every six months, while Debian releases a new version every two to three years and Community: Ubuntu has a large and active community

Reviewer: I'd say Fedora meets those criteria and none of them are related to running or building Kroxylicious. So I don't think its appropriate for Kroxylicious to recommend a distribution.

3. Launch the Linux distribution and Configure the Linux environment: After launching the Linux distribution, you can configure the environment by creating a user account and setting the password.

With these steps, you should now have WSL installed and configured on your Windows system. 

For further assistance:https://learn.microsoft.com/en-us/windows/wsl/install

### Build the project 

1. Open the WSL window and update the packages using

```bash
  sudo apt update
  sudo apt upgrade
```
2. Install Latest Java version by typing 
```bash
java
```
and Choose the version (preferably v18), it should look like:
```bash
sudo apt install openjdk-18-jre-headless
```
3. Check whether Java is installed correctly or not using:
```bash
java --version
```

> It should give the description of jdk package installed.

Reviewer: I think this is still part of setting up WSL rather than building the project.

4. Navigate to kroxylicious folder using :
```bash
 cd kroxylicious
 ```

5. Build it now using:
```bash
mvn clean install

> ```

Reviewer: We are back to the standard build instructions so maybe better to just link to the standard build steps from the WSL setup.

Author: i think it would be good let me correct it

Author: please check now
